apps/web/src/components/templates/notification-setting-form/NotificationSettingsForm.tsx

```diff
@@ -41,17 +41,17 @@ export const NotificationSettingsForm = ({
   });
 
   useEffect(() => {
+    function selectFirstGroupByDefault() {
+      setTimeout(() => {
+        setValue('notificationGroup', groups[0]._id);
+      }, 500);
+    }
+
     const group = getValues('notificationGroup');
     if (groups?.length && !editMode && !group) {
       selectFirstGroupByDefault();
     }
-  }, [groups, editMode]);
-
-  function selectFirstGroupByDefault() {
-    setTimeout(() => {
-      setValue('notificationGroup', groups[0]._id);
-    }, 500);
-  }
+  }, [groups, editMode, getValues, setValue]);
```

Comment:
Some of this were quick fixes for the error checks given by the validator. I reviewed some of the cases manually to be able to completing the passing of some test suites. As I haven't yet fulfilled all the suites this might be part of the ones to review.